packages/menu/src/useMenu.ts

```diff
@@ -341,25 +366,25 @@ export const useMenu = <T extends HTMLElement = HTMLElement, M extends HTMLEleme
 
         const type = StateChangeTypes[key === KEYS.ESCAPE ? 'MenuKeyDownEscape' : 'MenuKeyDownTab'];
 
-        if (KEYS.ESCAPE === key) {
-          focusTriggerRef.current = true;
-        }
+        // Skip focus return on TAB key. Focus should go to the next element in the tab order.
+        returnFocusToTrigger(KEYS.TAB === key);
 
         closeMenu(type);
       }
     },
-    [closeMenu, focusTriggerRef]
+    [closeMenu, returnFocusToTrigger]
   );
 
   const handleMenuBlur = useCallback(
     (event: MouseEvent) => {
       const path = event.composedPath();
 
       if (!path.includes(menuRef.current!) && !path.includes(triggerRef.current!)) {
+        returnFocusToTrigger();
         closeMenu(StateChangeTypes.MenuBlur);
       }
     },
-    [closeMenu, menuRef, triggerRef]
+    [closeMenu, menuRef, returnFocusToTrigger, triggerRef]
```

Comment:
I just realized this and not sure if it's represented in the use cases here, but what happens in a situation where. a consumer controls `isExpended` and selecting a selectable item won't immediately close the menu?

https://garden.zendesk.com/components/menu#grouped-items

Reply:
Bringing up because I see "controlledIsExpanded" is checked as a `skip` condition for `returnFocusToTrigger`, but there might be this other complication when expansion is controlled.

Again, maybe thats covered, I just want to confirm :)

Reply:
Maybe a `isExpanedControlled` additional check as a skip condition would be sufficient? 👀